src/main/java/com/crosswordservice/controller/CrosswordController.java

```diff
@@ -0,0 +1,142 @@
+        Question quest4 = new Question(config.getId(),"What is the abbreviation of the computing unit in a computer?","CPU");
+        questions.add(quest1);
+        questions.add(quest2);
+        questions.add(quest3);
+        questions.add(quest4);
+        questionRepository.saveAll(questions);
+
+        return questions;
+    }
+
+    /**
+     * Adds a configuration
+     * @param configuration configuration which will be added
+     * @return added configuration
+     */
+    @PostMapping("/configuration")
+    public Configuration saveConfiguration(@RequestBody Configuration configuration){
+        configurationRepository.save(configuration);
+        return configuration;
+    }
+
+    /**
+     * Adds questions to a configuration.
+     * @param questions question which will be added to the configuration
+     * @param name configuration name
+     * @return added questions
+     */
+    @PostMapping("/questions/{name}")
+    public List<Question> saveAllQuestions(@RequestBody List<Question> questions, @PathVariable String name) {
+        Configuration config = configurationRepository.findByName(name);
+        questions.forEach(question -> {
+            question.setInternalId(config.getId());
+        });
+        questionRepository.saveAll(questions);
+        return questions;
+    }
+
+    /**
+     * Updates the questions of a configuration.
+     * @param questions questions with an answer
+     * @param name configuration name
+     * @return updated questions.
+     */
+    @PutMapping("/questions/{name}")
+    public List<Question> updateAllQuestions(@RequestBody List<Question> questions, @PathVariable String name){
+        Configuration config = configurationRepository.findByName(name);
+        if(config == null){
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND,"id not found");
+        }
+        questionRepository.deleteByInternalId(config.getId());
+        questions.forEach(question -> {
+            question.setInternalId(config.getId());
+        });
+        questionRepository.saveAll(questions);
+        return questions;
+    }
+
+    /**
+     * Deletes a question with the given id.
+     * @param id Id of a question
+     * @return deleted question
+     */
+    @DeleteMapping("/question/{id}")
+    public Question removeQuestion(@PathVariable Long id){
+        Question question = questionRepository.getReferenceById(id);
+        questionRepository.deleteById(id);
+        return question;
+    }
+
+    /**
+     * Returns all questions
+     * @return all questions
+     */
+    @GetMapping("/questions")
+    public List<Question> getAllQuestions() {
+        return (List<Question>) questionRepository.findAll();
+    }
+
+    /**
+     * Returns all questions of a configuration
+     * @param name Configuration name
+     * @return list of questions of a configuration
+     */
+    @GetMapping("/questions/{name}")
+    public List<Question> getAllQuestions(@PathVariable String name) {
```

Comment:
I would prefer to give the method a other name than the above.
```suggestion
    public List<Question> getAllQuestionsByConfigurationName(@PathVariable String name) {
```